Add guide explaining default axis names (#942)
napari/docs: docs/guides/axis-names.md

---
jupytext:
  text_representation:
    extension: .md
    format_name: myst
    format_version: 0.13
    jupytext_version: 1.10.3
kernelspec:
  display_name: Python 3
  language: python
  name: python3
---

(axis-names)=

# Axis names in napari

```{warning}
The napari team is actively working to improve the consistency and user
experience of axis names and other axis metadata. This guide should be up to

> date, but things may be in a confusing state. It's not you, it's us! 😅 In

Reviewer: Do we usually put emojis in our docs? I don't think we do (aside from release notes), so I'd remove him here.

Author: boo.

Author: ```suggestion
date, but things may be in a confusing state. It's not you, it's us! In
```

summary:

- **Before 0.6.0,** napari's `Viewer.dims` held axis names, and these were
  mapped to layer axes purely based on order. Layer axes did not have names.
- **In 0.6.x,**, layers gained the ability to hold axis names and axis units,
  which was important for [reader plugins](contributions-readers) to be able
  to provide this information. However, napari did not use that information for
  anything: it was up to the user to propagate it to `Viewer.dims`.
- **In 0.7.x,** layer axis *units* propagate to `Viewer.dims` automatically,

>   but layer names are still ignored by default.

Reviewer 2: ```suggestion
  but layer axis *names* are still ignored by default.
```

- [**In the hopefully near future,**](active-roadmap), layer axis names will be
  automatically combined to provide the viewer axis names, which will be the
  union of all layer axes.
```

napari started out as a tool to look at NumPy arrays. As such, it mimics some

> behavior when mixing arrays of different dimension, known as

Reviewer: ```suggestion
behavior when mixing arrays of different dimensions, known as
```

[broadcasting](https://numpy.org/doc/stable/user/basics.broadcasting.html).

Specifically, arrays of mixed dimensionality are *right-aligned*. In order to

> make array shapes match across different elements of napari, napari 0.7.0

Reviewer 2: I don't have a suggestion, but I do find this to be unclear. Do you mean array dims instead of shapes?

Reviewer 3: I think `array axes` would be clearer?
```suggestion
Specifically, arrays of mixed dimensionality are *right-aligned*. In order to
make array axes match across different elements of napari, napari 0.7.0
```

started using *negative indexing*, i.e. indexing from the right end of a list

> or tuple, for its axes.

Reviewer: ```suggestion
or tuple, for its axes labels.
```

Suppose we have a time lapse of neuronal activity, with a 3D image with
dimensions T, Y, and X, a set of 2D masks of dimensions Y, X, and a set of
2D centroid coordinates, with dimensions Y, X. Before napari 0.7.0, the default

> dimension names would have been:

Reviewer: ```suggestion
axis labels would have been:
```
